src/colors.js

```diff
@@ -0,0 +1,57 @@
+import {color} from 'chart.js/helpers';
+import {getGradientData} from './helpers';
+
+const rgbs = (c) => Math.round(c._rgb.a * 255) << 24 | c._rgb.r << 16 | c._rgb.g << 8 | c._rgb.b;
+// IEC 61966-2-1:1999
+const toRGBs = (l) => l <= 0.0031308 ? l * 12.92 : Math.pow(l, 1.0 / 2.4) * 1.055 - 0.055;
+// IEC 61966-2-1:1999
+const fromRGBs = (srgb) => srgb <= 0.04045 ? srgb / 12.92 : Math.pow((srgb + 0.055) / 1.055, 2.4);
+
+function interpolate(percent, startColor, endColor) {
+  const startRGBs = rgbs(startColor.color);
+  const startR = fromRGBs(((startRGBs >> 16) & 0xff) / 255);
+  const startG = fromRGBs(((startRGBs >> 8) & 0xff) / 255);
+  const startB = fromRGBs((startRGBs & 0xff) / 255);
+
+  const endRGBs = rgbs(endColor.color);
+  const endR = fromRGBs(((endRGBs >> 16) & 0xff) / 255);
+  const endG = fromRGBs(((endRGBs >> 8) & 0xff) / 255);
+  const endB = fromRGBs((endRGBs & 0xff) / 255);
+
+  const r = Math.round(toRGBs(startR + percent * (endR - startR)) * 255);
+  const g = Math.round(toRGBs(startG + percent * (endG - startG)) * 255);
+  const b = Math.round(toRGBs(startB + percent * (endB - startB)) * 255);
+
+  return color({r, g, b, a: 1});
```

Comment:
Not sure if alpha channel can be used with gradients, but if it can, it should be interpolated too.

Something like 
```js
const a = startB + percent * (endB - startB)
```

Another note, can do the calculation while constructing the result object (to avoid creating intermediate variables):
```js
return color({
  r: Math.round(toRGBs(startR + percent * (endR - startR)) * 255),
  ...
});
```

Reply:
> Not sure if alpha channel can be used with gradients, but if it can, it should be interpolated too.

If you see some previous commits, I was calculating but if you apply the alpha, it seems the legend is not showing it. I'll do some additional tests.


> Another note, can do the calculation while constructing the result object (to avoid creating intermediate variables):

OKI!!!

Reply:
I have done some tests but I think it's not correct to calculate the interpolation on the alpha, based on percentage of the value, because when the value is small, the alpha is close to zero and the color is not showed. Maybe the max of alphas could be calculated. If you agree, I'll remain at `a: 1`

Reply:
I was wrong... added interpolation on alpha as well.

Reply:
Thanks @kurkle because you gave me the input to change my lib as well to be more performed!!